sys/include/net/gnrc/ipv6.h

```diff
@@ -11,10 +11,63 @@
  * @ingroup     net_gnrc
  * @brief       GNRC's IPv6 implementation
  *
- * The IPv6 control thread understands messages of type
- *
- *  * @ref GNRC_NETAPI_MSG_TYPE_RCV, and
- *  * @ref GNRC_NETAPI_MSG_TYPE_SND,
+ * This module is for usage with the @ref net_gnrc_netapi
+ *
+ * # NETAPI command documentation
+ *
+ * ## `GNRC_NETAPI_MSG_TYPE_RCV`
+ *
+ * @ref GNRC_NETAPI_MSG_TYPE_RCV expects a @ref net_gnrc_pkt (referred to as
+ * "packet" in the following) in receive order (payload first, headers in
+ * decreasing order by layer). It should at least contain a link-layer payload.
+ * If it contains a @ref net_gnrc_netif_hdr it will be taken into account.
+ *
+ * If the link-layer payload is not an IPv6 packet (starting with a 0x6
+ * half-byte) or the module is not able to parse it correctly it will be
+ * dropped. If the packet can't be forwarded (either to another node or an
+ * application) it will be dropped. Otherwise, there are two scenarios of
```

Comment:
Unclear: netapi is not forwarding to nodes, but stack-internal. Do you mean: "forwarded up or down the stack"?

Reply:
I'm assuming that "If the packet can't be forwarded" refers to the white/blacklisting checks in `gnrc.ipv6.c:_receive()`– if a packet's source address is blacklisted/not whitelisted, the packet **cannot** be forwarded. It doesn't matter if the ipv6 module is the module doing the actual forwarding; the ipv6 module is where this decision is made.
In case this assumption is correct, I'm happy with the sentence as it is– it states what happens without going into unnecessary detail.

Reply:
This does not refer to netapi but the packet level. If the packet can't be forwarded (i.e. there is no FIB entry) then it is dropped (with an error message if `gnrc_icmpv6_error` is compiled in). "Forwarding to an application" is meant in the way of resolving the `(nettype, demux_ctx)` tuple. So @Lotterleben was right, but not only with regards to "firewalling". I will however replace `application` with `upper layer` ;-).